src/cohortManager.js

```diff
@@ -0,0 +1,153 @@
+class CohortDetainment {
+  constructor() {
+    this.cohortManager = []
+    this.idCount = 1
+  }
+
+  createCohort(cohortName) {
+    if (cohortName.length < 1) {
+      throw new Error('Empty string is not valid!')
+    }
+    this.cohortManager.forEach((teamName) => {
+      for (const key in teamName) {
+        if (key === cohortName) {
+          throw new Error('Cohort already Exists!')
+        }
+      }
+    })
+    const newCohort = { [cohortName.toLowerCase()]: [] }
+    this.cohortManager.push(newCohort)
+    return newCohort
+  }
+
+  searchForCohort(cohortName) {
+    const resultOfFind = this.cohortManager.find((obj) =>
+      obj.hasOwnProperty(cohortName)
+    )
+    return resultOfFind
+  }
+
+  addStudent(cohortName, newStudentFirstName, newStudentLastName) {
+    this.cohortManager.forEach((team) => {
+      for (const key in team) {
+        for (let i = 0; i < team[key].length; i++) {
+          if (
+            team[key][i].firstName === newStudentFirstName &&
+            team[key][i].lastName === newStudentLastName
+          ) {
```

Comment:
What is the purpose of this bit? Might you be able to use a find function for this?

Reply:
potentially, but i was looking at objects within objects and arrays so could more easily map it in my head like this